src/gladvent/internal/cmd/new.gleam

```
@@ -41,23 +46,38 @@ fn create_input_file(
 ) -> fn() -> Result(String, Err) {
   fn() {
     let input_path = input.get_file_path(ctx.year, ctx.day, kind)
-    simplifile.create_file(input_path)
-    |> result.map_error(handle_file_open_failure(_, input_path))
+    case kind, ctx.fetch_input {
+      input.Puzzle, True -> {
+        use content <- result.try(download_input(ctx))
```

💭  in order to avoid downloading input that has already been fetched, we may to to rework this function somewhat to always attempt to create the file, and only download and fill it with input if we have successfully created it or we find a way to guarantee that the file is empty (I assume simplifies has a function we can leverage for this)

---

As the `--fetch` is a flag to new, we could probably assume that people would run `new` only once. Wouldn't it then be simpler to just assume we don't want to do anything if the file already exists, and then throw the same `FileAlreadyExists` error as when trying to create it? Checking for emptiness seems a bit excessive.

---

Yeah that works too! in that case we create as normal and let that fail/succeed and then only write the contents if the create succeeded

---

I have explicitly checked for file existence to avoid having to create an empty file (see my last commit), but you'd maybe have preferred to always call the `create_file` function and then call `download_input` only if `create_file` succeeded (thus the file wasn't there before) and we fill the conditions (input kind and flag).
I can rewrite that if you think it's best and/or easier to maintain.